### PR TITLE
HOOK-3244 Introduce action

### PR DESCRIPTION
**Pull request by romansaul**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    - name: Get credentials
+      uses: leanix/secrets-action@master
+      with:
+        secret-store-credentials: ${{ secrets.INJECTED_SECRET_STORE_CREDENTIALS }}
+        
+    - name: Build and push docker image
+      uses: leanix/release-docker-image-action@master
+      with:
+        name: leanix/integration-api-default-config-action
```
Comment (dan-leanix) [inline .github/workflows/build.yml:29]:
```suggestion
```
You can keep it, but per default you don't need it as the repo name does not contain `leanix`. For instance, `leanix-integration-api` needs the `with: name` option.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    apt-get install -y ca-certificates curl apt-transport-https lsb-release gnupg jq && \
+    apt-get clean
+
+# Install azure-cli
+RUN curl -sL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor | tee /etc/apt/trusted.gpg.d/microsoft.asc.gpg > /dev/null && \
+    AZ_REPO=$(lsb_release -cs) && \
+    echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" | tee /etc/apt/sources.list.d/azure-cli.list && \
+    apt-get update && \
+    apt-get install azure-cli && \
+    apt-get clean
```
Comment (dan-leanix) [inline Dockerfile:14]: I would combine both RUN steps into one.

**Review by dan-leanix — CHANGES_REQUESTED** [review]

File: entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+fi
+if [[ -z $AZURE_TENANT_ID ]]; then
+  export AZURE_TENANT_ID=$ARM_TENANT_ID
+fi
+if [[ -z $AZURE_SUBSCRIPTION_ID ]]; then
+  export AZURE_SUBSCRIPTION_ID=$ARM_SUBSCRIPTION_ID
+fi
+
+echo "Login to Azure..."
+az login --service-principal -u $ARM_CLIENT_ID -p $ARM_CLIENT_SECRET --tenant $ARM_TENANT_ID
```
Comment (rwe17) [inline entrypoint.sh:34]: Why is `AZURE_CLIENT_ID` exported and only ARM_CLIENT_SECRET is used inside the script? Do i oversee something?

File: entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+declare -A REGION_IDS
+REGION_IDS["westeurope"]="eu"
+REGION_IDS["eastus"]="us"
+REGION_IDS["canadacentral"]="ca"
+REGION_IDS["australiaeast"]="au"
+REGION_IDS["germanywestcentral"]="de"
+REGION_IDS["horizon"]="horizon" # edge case
+
+# The file containing the default integration-api config from the calling repository
+NEW_CONFIG_LOCATION=/github/workspace/${INPUT_DEFAULT_CONFIG_FILE}
```
Comment (rwe17) [inline entrypoint.sh:14]: Is `/github/workspace/` a directory, we need to create later in our leanix-integration-api project?

**Review by rwe17 — APPROVED** [review]
I still have some questions, but i assume this PR is a good starting point to start the remaining configuration on our side and do some tests.

**Review by oaquijad — APPROVED** [review]
This is the first step in order to create a new integration-api default configuration using this action, right? (Because it is still a TODO to actually create or update this default configuration).

File: README.md [hunk 5]
```diff
@@ -1 +1,32 @@
...
+
+## Usage
+
+This action reads the default configuration from a file and calls the Integration-API to provision the provided config.
+Use in advance the provided [leanix/secrets-action](https://github.com/leanix/secrets-action) to inject the required secrets.
+
+A simple provision step in the connector's workflow would look like this:
+```yaml
+- name: Provision default config
+  uses: leanix/integration-api-default-config-action@master
```
[the remark this answers is not in the document]
Comment (romansaul) [reply README.md:13]: Good catch!

File: entrypoint.sh (hunk 4 above)
Comment (romansaul) [reply entrypoint.sh:14]: The action is used in the repo of the connector. All files from that repo are mounted at /github/workspace in the container which is the runtime of the action.

**Comment by romansaul** [comment]
> This is the first step in order to create a new integration-api default configuration using this action, right? (Because it is still a TODO to actually create or update this default configuration).

Yes, the TODOs should be replaced in the course of working on the sibling Sailors ticket https://leanix.atlassian.net/browse/INTG-2256

File: entrypoint.sh (hunk 3 above)
Comment (romansaul) [reply entrypoint.sh:34]: Yes, I'm also wondering. I think line 19-31 are a copy&paste error and can be removed.

**Review by rwe17 — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (romansaul) [reply Dockerfile]: Done

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (romansaul) [reply README.md]: Fixed.

File: entrypoint.sh (hunk 3 above)
Comment (romansaul) [reply entrypoint.sh:20]: Removed.

**Review by dan-leanix — APPROVED** [review]

**Pull request merged**